pre-receive-hooks/commit-current-user-check.sh

```diff
@@ -0,0 +1,84 @@
+#!/usr/bin/env bash
+#
+# Pre-receive hook that will reject all pushes where author or committer are not the current user.
+#
+# Pre-requisites for the users.
+# They must have:
+# * git config --global user.email set to an email address
+# * That email address must be set as a public email address in GitHub Enterprise
+# * git config --global user.name must be set to GitHub Enterprise login name
+
+# If we are on the GitHub Web interface then we don't need to bother to validate the commit user
+if [[ "${GITHUB_VIA}" == "pull request merge button" ]] || \
+   [[ "${GITHUB_VIA}" == "blob edit" ]]; then
+   exit 0
+fi
+
+# Set up a user token (attached to a non expiring account) that can just read public email addresses.
+TOKEN=USER:TOKEN
+
+# We set the address of the GHE Instance here
+GHE_URL=https://GHE-INSTANCE
```

Comment:
You cannot use localhost as that is not defined in the shell used by the pre-receive hook. The variable $HOSTNAME cannot be used either.

GHE takes the server name and alters it. so what starts out as ```ghe.example.github.com``` ends up in $HOSTNAME as ```ghe-example-github-com-primary``` . Although $HOSTNAME equates to 127.0.01, the API is not accessible through that IP address from the hook environment. It returns no data.

Also the error message at:
```echo "Please set public email address at {GHE_URL}/settings/profile."```
would become
``` Please set public email address at https://ghe-example-github-com-primary/settings/profile.```
rather than
``` Please set public email address at https://ghe.example.github.com/settings/profile.```